ddclient

```diff
@@ -1475,8 +1475,11 @@ sub test_possible_ip {
 
     {
         local $opt{'use'} = 'if';
-        foreach my $if (grep {/^[a-zA-Z]/} `ifconfig -a`) {
-            $if =~ s/:?\s.*//is;
+        my @ifs = map({ /^[^\s:]*:\s*([^\s:]+):/ ? $1 : () } `ip -o link show 2>/dev/null`);
```

Comment:
This fails for VLANS where ip returns the interface with a @ sign between the name and the base interface that the VLAN is attached to.  Testing your code I get the following output...

```
eth0
eth1
eth2
eth3
eth1.100@eth1
eth1.200@eth1
eth1.300@eth1
```

So for each of the VLANS the regex needs to terminate at the @ symbol (or the colon).  I'm not very good with regex so don't have a quick fix for you.

Reply:
Wow, that's poor design. Why does it show `@eth1` there when that's not actually part of the interface name?  :angry:

Fixed, thanks for bringing it up.